app/helpers/react_on_rails_helper.rb

```diff
@@ -49,7 +49,7 @@ def react_component(component_name, props = {}, options = {})
     # Create the HTML rendering part
     if prerender
       render_js_expression = <<-JS
-          renderReactComponent(this.#{reactComponent}, #{props.to_json})
+        this.React.renderToString(#{render_js_react_element(reactComponent, props.to_json)});
```

Comment:
Let's fix the lines above changing:
```
     dataVariable = "__#{component_name.camelize(:lower)}Data#{@react_component_index}__"
     reactComponent = component_name.camelize
```
to
```
     data_variable = "__#{component_name.camelize(:lower)}Data#{@react_component_index}__"
     react_component_name = component_name.camelize
```
That way, it's clear which vars are ruby and which are javascript. and change react_component to react_component_name.

Let's break this into a variable: 
```
var reactComponent = #{render_js_react_element(react_component_name, props.to_json)};
```